src/Utilities/Compiler/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -287,6 +287,57 @@ public static Accessibility DetermineMinimalAccessibility(this ITypeSymbol typeS
             return typeSymbol.Accept(MinimalAccessibilityVisitor.Instance);
         }
 
+        public static bool HasAnyCollectionCountProperty(this ITypeSymbol invocationTarget, WellKnownTypeProvider wellKnownTypeProvider)
```

Comment:
Please feel free to suggest a better name, I am not entirely satisfied with this one but couldn't find any better (with a reasonable length xD).

Reply:
You can consider adding a doc comment on this method in your follow-up PR that uses this method.